### DISCUSSION TITLE
Easy way to hide a table or list if a $param is not set?

### DISCUSSION THREAD

**Discussion by shah** [sqlpage/SQLPage#185] [Q&A] [answered]
I'm creating a drill-down page where I show a "master" table with hyperlinks that self-reference the page using `?drill-param=X` links and a "detail" table which shows the values of the second level when the param is set. I've been able to create two table components and everything works great in SQLPage except when there's no "selected" master item it shows a blank table.

I've seen the discussion / ticket (#142)  to create a subquery with `HAVING` clause but in my case all I need to do is check a single parameter so I wasn't sure how to proceed. 

Thoughts?

See:

![image](https://github.com/lovasoa/SQLpage/assets/47497/08904a8d-dd7a-496f-9b5a-f33e3f3b4872)

**Comment by lorefnon** [comment, marked as answer]
We can do conditional selects for the detail table: 

```sql
select 
    'table' as component
where coalesce($drill_param, '') <> '';
select 
    name,
    dob
from users
where coalesce($drill_param, '') <> '' and users.cohort = $drill_param;
```

This approach requires putting the condition in both places - the component selection as well as data selection.

Alternative, is to use a card with embed: 

```sql
select 
    'card' as component,
    1      as columns;
select 
    'Users' as title,
    '/users_table.sql?_sqlpage_embed' as embed
where coalesce($drill_param, '') <> '';
``` 

Where the users_table.sql has the SQL for rendering the users table. It will be fetched and injected into the page after load if the $drill_param variable is not empty.

**Reply by shah** [reply]
The `where coalesce($drill_param, '') <> ''` in the master table worked like a charm. Thanks!
